
From: Carlos Eduardo Medaglia Dyonisio <medaglia@undl.org.br>

This patch fixes troubles when compiling some applications that include
<linux/byteorder/little_endian.h>, like xmms.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/linux/types.h |   14 +++++++-------
 1 files changed, 7 insertions(+), 7 deletions(-)

diff -puN include/linux/types.h~fix-typesh include/linux/types.h
--- 25/include/linux/types.h~fix-typesh	2004-09-23 00:07:24.137527888 -0700
+++ 25-akpm/include/linux/types.h	2004-09-23 00:07:24.140527432 -0700
@@ -140,6 +140,13 @@ typedef unsigned long sector_t;
 #define pgoff_t unsigned long
 #endif
 
+#endif /* __KERNEL_STRICT_NAMES */
+
+/*
+ * Below are truly Linux-specific types that should never collide with
+ * any application/library that wants linux/types.h.
+ */
+
 #ifdef __CHECKER__
 #define __bitwise __attribute__((bitwise))
 #else
@@ -153,13 +160,6 @@ typedef __u32 __bitwise __be32;
 typedef __u64 __bitwise __le64;
 typedef __u64 __bitwise __be64;
 
-#endif /* __KERNEL_STRICT_NAMES */
-
-/*
- * Below are truly Linux-specific types that should never collide with
- * any application/library that wants linux/types.h.
- */
-
 struct ustat {
 	__kernel_daddr_t	f_tfree;
 	__kernel_ino_t		f_tinode;
_
